### PR TITLE
refactor(types): enable noImplicitAny on transaction-runner.ts & test/transaction-runner.ts

### PR DESCRIPTION
**Pull request by laljikanjareeya**
Fixes #<issue_number_goes_here> (it's a good idea to open an issue first for discussion)

- [ ] Tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: test/transaction-runner.ts [hunk 1]
```diff
@@ -332,12 +333,12 @@ describe('TransactionRunner', () => {
       describe('transaction requests', () => {
         const CONFIG = {};
 
-        let callbackStub;
+        let callbackStub: sinon.SinonSpy<Any, Any>;
 
         beforeEach(() => {
           callbackStub = sandbox.spy();
```
Comment (praveenqlogic) [inline test/transaction-runner.ts:355]: can you give a try with function/sinon.SinonSpy here!

File: test/transaction-runner.ts [hunk 2]
```diff
@@ -46,14 +49,14 @@ describe('TransactionRunner', () => {
...
   };
 
-  // tslint:disable-next-line no-any variable-name
-  let Runner;
-  // tslint:disable-next-line no-any variable-name
-  let TransactionRunner;
-  // tslint:disable-next-line no-any variable-name
-  let AsyncTransactionRunner;
+  // tslint:disable-next-line variable-name
+  let Runner: Any;
```
Comment (praveenqlogic) [inline test/transaction-runner.ts]: import * as trTypes from '../src/transaction-runner';
let Runner:typeof trTypes.Runner;
let TransactionRunner: typeof trTypes.TransactionRunner;
let AsyncTransactionRunner: typeof trTypes.AsyncTransactionRunner;

File: test/transaction-runner.ts [hunk 3]
```diff
@@ -75,17 +78,17 @@ describe('TransactionRunner', () => {
...
   describe('Runner', () => {
-    // tslint:disable-next-line no-any variable-name
-    let ExtendedRunner;
+    // tslint:disable-next-line variable-name
+    let ExtendedRunner: Any;
 
-    let runFn;
-    let runner;
+    let runFn: sinon.SinonStub<Any, Any>;
+    let runner: Runner<Any>;
```
Comment (praveenqlogic) [inline test/transaction-runner.ts]: let runFn:sinon.SinonStub;
    let runner:trTypes.Runner<void>;

File: test/transaction-runner.ts [hunk 4]
```diff
@@ -383,9 +384,9 @@ describe('TransactionRunner', () => {
           fakeError.code = status.ABORTED;
 
           fakeTransaction.request.onCall(0).callsFake(
-              (_, callback) => callback(fakeError));
+              (_: object, callback: Function) => callback(fakeError));
```
Comment (praveenqlogic) [inline test/transaction-runner.ts]: use {} instead of object type

File: test/transaction-runner.ts [hunk 5]
```diff
@@ -401,9 +402,9 @@ describe('TransactionRunner', () => {
           fakeError.code = status.UNKNOWN;
 
           fakeTransaction.request.onCall(0).callsFake(
-              (_, callback) => callback(fakeError));
+              (_: object, callback: Function) => callback(fakeError));
```
Comment (praveenqlogic) [inline test/transaction-runner.ts]: use {} instead of object type, follow the same!

File: test/transaction-runner.ts [hunk 6]
```diff
@@ -496,18 +497,18 @@ describe('TransactionRunner', () => {
           fakeError.code = status.UNKNOWN;
 
           const fakeData = [{a: 'b'}, {c: 'd'}, {e: 'f'}];
-          fakeData.forEach(data => goodStream.push(data));
+          fakeData.forEach((data: object) => goodStream.push(data));
```
Comment (praveenqlogic) [inline test/transaction-runner.ts]: const fakeData:Array<{}> = [{a: 'b'}, {c: 'd'}, {e: 'f'}];
remove object type from data

File: test/transaction-runner.ts (hunk 3 above)
Comment (laljikanjareeya) [reply test/transaction-runner.ts]: Here Runner is Type of `Runner<T>`

File: test/transaction-runner.ts (hunk 1 above)
Comment (laljikanjareeya) [reply test/transaction-runner.ts:355]: Done!

File: test/transaction-runner.ts (hunk 4 above)
Comment (laljikanjareeya) [reply test/transaction-runner.ts]: Done!

File: test/transaction-runner.ts (hunk 5 above)
Comment (laljikanjareeya) [reply test/transaction-runner.ts]: Done!

File: test/transaction-runner.ts (hunk 6 above)
Comment (laljikanjareeya) [reply test/transaction-runner.ts]: Done!

**Pull request closed without merging**